### PR TITLE
[FEATURE] Add Support for querying Avro Data in Kafka Connector

### PR DESCRIPTION
**Pull request by Jithendar12**
Support for querying Avro Data from Kafka.

Prerequisite: Define the Avro Schema in AWS Glue Schema Registry, Use Data Format as AVRO.
Note: Provide Glue Schema Name as Kafka Topic Name with same case .
example avro schema in glue :
```
{
    "type": "record",
    "name": "employee",
    "fields": [
        {
            "name": "id",
            "type": "int"
        },
        {
            "name": "name",
            "type": "string"
        }
    ]
}
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-kafka/src/main/java/com/amazonaws/athena/connectors/kafka/KafkaMetadataHandler.java [hunk 1]
```diff
@@ -419,29 +428,49 @@ private Schema getSchema(String glueRegistryName, String glueSchemaName) throws
...
-        schemaBuilder.addMetadata("dataFormat", topicSchema.getMessage().getDataFormat());
-
+        if (registryReader.getGlueSchemaType(glueRegistryName, glueSchemaName).equalsIgnoreCase(AVRO_DATA_FORMAT)) {
+            AvroTopicSchema avroTopicSchema = registryReader.getGlueSchema(glueRegistryName, glueSchemaName, AvroTopicSchema.class);
+            // Creating ArrowType for each fields in the topic schema.
+            // Also putting the additional column level information
+            // into the metadata in ArrowType field.
+            avroTopicSchema.getFields().forEach(it -> {
+                FieldType fieldType = new FieldType(
+                        true,
```
Comment (AbdulR3hman) [inline athena-kafka/src/main/java/com/amazonaws/athena/connectors/kafka/KafkaMetadataHandler.java:438]: are all types are nullable?

File: athena-kafka/src/main/java/com/amazonaws/athena/connectors/kafka/KafkaRecordHandler.java [hunk 2]
```diff
@@ -219,4 +251,90 @@ private void execute(
...
+    private void avroConsume(
+            BlockSpiller spiller,
+            ReadRecordsRequest recordsRequest,
+            QueryStatusChecker queryStatusChecker,
+            SplitParameters splitParameters,
+            Consumer<String, GenericRecord> kafkaAvroConsumer)
+    {
+        LOGGER.info("[kafka] {} Polling for data", splitParameters);
+        int emptyResultFoundCount = 0;
+        try {
```
Comment (AbdulR3hman) [inline athena-kafka/src/main/java/com/amazonaws/athena/connectors/kafka/KafkaRecordHandler.java:264]: It seems we are only using a try for the sake of closing [Consumer](https://kafka.apache.org/31/javadoc/org/apache/kafka/clients/consumer/Consumer.html) -- why not use [try with resource](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)?

File: athena-kafka/src/main/java/com/amazonaws/athena/connectors/kafka/KafkaMetadataHandler.java (hunk 1 above)
Comment (Jithendar12) [reply athena-kafka/src/main/java/com/amazonaws/athena/connectors/kafka/KafkaMetadataHandler.java:438]: Hi @AbdulR3hman ,
Yes, all data types that we are supporting for this connector can have null values in Kafka topic messages.

File: athena-kafka/src/main/java/com/amazonaws/athena/connectors/kafka/KafkaRecordHandler.java (hunk 2 above)
Comment (Jithendar12) [reply athena-kafka/src/main/java/com/amazonaws/athena/connectors/kafka/KafkaRecordHandler.java]: Addressed

**Review by AbdulR3hman — APPROVED** [review]

**Pull request merged**